docs/api/expect.md

```diff
@@ -1136,6 +1136,32 @@ If the value in the error message is too truncated, you can increase [chaiConfig
   })
   ```
 
+## expect.unreachable
+
+- **Type:** `(message?: string) => never`
+
+  This method is used to assert that a line should never be reached.
+
+  For example, if you want to test a function that should throw an error, we can add this assertion after the function call.
+
+  ```ts
+  import { expect, test } from 'vitest'
+
+  function throwError() {
+    throw new Error('Oops')
+  }
+
+  test('throwError throws an error with a message', () => {
+    try {
+      throwError()
+      expect.unreachable('I will not run')
+    }
+    catch (err: any) {
+      expect(err.message).toBe('Oops')
```

Comment:
I feel like this example is too simplistic, because it's better to rewrite it into `expect(() => throwError()).toThrowError('Oops')`

Maybe we can add more checks to justify this pattern?

Like:

```ts
expect(err).toBeInstanceOf(Error)
expect(err.stack).toContain('throwError')
```

Reply:
I agree with you. I'll add them for now, but we better find another example or just leave it without examples.

Reply:
👀 Just found some `assert.unreachable` calls in SvelteKit codebase, I will create a new example.